### PR TITLE
feat: add utxo set

### PR DESCRIPTION
**Pull request by sczembor**
## Summary by Sourcery

Implement a UTXO set in the bitcoin_executor by defining OutPoint and Info types, integrating a UTXO table into State, and adding functions to add, spend (with coinbase maturity enforcement), and query UTXOs.

New Features:
- Introduce a utxo module with OutPoint and Info structs and associated constructors and accessors
- Add add_utxo, spend_utxo (including coinbase maturity check), and utxo_exists functions to manage the UTXO set
- Incorporate a UTXO table field into State and initialize it in the module init function

Enhancements:
- Define ECoinbaseNotMature error constant for aborting on immature coinbase spends
- Update module init to create and share the UTXO-enabled State object

Tests:
- Add unit tests covering UTXO addition, duplicate insertion failure, spending, and coinbase maturity enforcement

### CODE REVIEW

File: bitcoin_executor/sources/utxo.move [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+public struct OutPoint has copy, drop, store {
+    tx_id: vector<u8>,
+    vout: u32,
+}
+
+public struct Info has copy, drop, store {
+    height: u64, // The height of the block containing the UTXO.
+    is_coinbase: bool, // Whether the UTXO is from a coinbase transaction or not.
+    value: u64, // Amount in satoshis
+    script_pub_key: vector<u8>, // The locking script
```
[the remark this answers is not in the document]
Comment (sczembor) [reply bitcoin_executor/sources/utxo.move:12]: `script_pub_key` is very commonly used in bitcoin. `scirpt_pub_key` = locking, `script_sig` = unlocking

File: bitcoin_executor/sources/utxo.move [hunk 2]
```diff
@@ -0,0 +1,44 @@
+module bitcoin_executor::utxo;
+
+public struct OutPoint has copy, drop, store {
```
[the remark this answers is not in the document]
Comment (sczembor) [reply bitcoin_executor/sources/utxo.move:3]: hm, OutPoint is again commonly used BTC https://btcinformation.org/en/glossary/outpoint. I like `OutPoint` and `Data` for the `utxo_set<OutPoint, Data`

File: bitcoin_executor/sources/utxo.move [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+public struct Info has copy, drop, store {
+    height: u64, // The height of the block containing the UTXO.
+    is_coinbase: bool, // Whether the UTXO is from a coinbase transaction or not.
+    value: u64, // Amount in satoshis
+    script_pub_key: vector<u8>, // The locking script
+}
+
+public fun new_outpoint(tx_id: vector<u8>, vout: u32): OutPoint {
```
[the remark this answers is not in the document]
Comment (sczembor) [reply bitcoin_executor/sources/utxo.move:15]: this is the `key` not the UTXO itself

File: bitcoin_executor/sources/utxo.move (hunk 2 above)
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply bitcoin_executor/sources/utxo.move:3]: Ok
